src/entry_point.rs

```diff
@@ -330,8 +330,9 @@ pub unsafe extern "C" fn rust_start(app_start: usize, stacktop: usize, app_heap_
     );
 
     // Zero .bss (specified by the linker script).
-    let bss_end = layout_header.bss_start + layout_header.bss_size; // 1 past the end of .bss
-    for i in layout_header.bss_start..bss_end {
+    let bss_start = stacktop + layout_header.bss_start;
+    let bss_end = stacktop + layout_header.bss_start + layout_header.bss_size; // 1 past the end of .bss
```

Comment:
Why not use `bss_start + layout_header.bss_size`?

Reply:
Yes, that's right. Will fix.

Reply:
This still doesn't seem fixed.